### PR TITLE
ovl_en_mm2 OK

### PR DESCRIPTION
**Pull request by emilybrooks**

### CODE REVIEW

File: src/overlays/actors/ovl_En_Mm2/z_en_mm2.c [hunk 1]
```diff
@@ -43,33 +48,292 @@ static ColliderCylinderInit D_80AAFAC0 = {
...
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mm2/func_80AAEE50.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mm2/func_80AAEF70.s")
+typedef struct {
+    /* 0x00 */ AnimationHeader* animation;
+    /* 0x04 */ f32 playSpeed;
+    /* 0x08 */ u8 mode;
+    /* 0x0C */ f32 morphFrames;
+} unknownStruct; // size = 0x10
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Mm2/z_en_mm2.c:57]: a struct for animation entries like this is pretty common. you can call it `Mm2AnimationEntry` or something along those lines

File: src/overlays/actors/ovl_En_Mm2/z_en_mm2.c [hunk 2]
```diff
@@ -43,33 +48,292 @@ static ColliderCylinderInit D_80AAFAC0 = {
...
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mm2/EnMm2_Init.s")
+static unknownStruct sAnimations[] = {
+    0x06000718, 1.0f, 0, -7.0f, 0x06006940, -1.0f, 2, -7.0f,  0x06006C50, 1.0f, 0, -7.0f,  0x06006940, 1.0f, 2, -7.0f,
+    0x06000468, 1.0f, 0, -7.0f, 0x060073A0, 1.0f,  0, -12.0f, 0x06008060, 1.0f, 0, -12.0f,
+};
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Mm2/z_en_mm2.c:62]: I would wrap each entry in { } so its clear which one is which

File: src/overlays/actors/ovl_En_Mm2/z_en_mm2.c [hunk 3]
```diff
@@ -43,33 +48,292 @@ static ColliderCylinderInit D_80AAFAC0 = {
...
+                         sAnimations[animationIndex].mode, phi_f0);
+    }
+    *previousAnimation = animationIndex;
+}
+
+void func_80AAEF70(EnMm2* this, GlobalContext* globalCtx) {
+    if ((gSaveContext.eventChkInf[9] & 0xF) != 0xF) {
+        this->actor.textId = 0x6086;
+        return;
+    }
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Mm2/z_en_mm2.c:93]: at first glance it seems like this function can be improved with else ifs and elses. might be overlooking something tho

File: src/overlays/actors/ovl_En_Mm2/z_en_mm2.c [hunk 4]
```diff
@@ -43,33 +48,292 @@ static ColliderCylinderInit D_80AAFAC0 = {
...
+    if (gSaveContext.infTable[23] & 0x8000) {
+        if (gSaveContext.eventInf[1] & 1) {
+            this->actor.textId = 0x6082;
+            return;
+        }
+        if (gSaveContext.timer2State) {
+            this->actor.textId = 0x6076;
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Mm2/z_en_mm2.c:100]: timer state isnt a bool i dont think, prob makes more sense to keep !=0

File: src/overlays/actors/ovl_En_Mm2/z_en_mm2.c [hunk 5]
```diff
@@ -43,33 +48,292 @@ static ColliderCylinderInit D_80AAFAC0 = {
...
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 21.0f);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &D_06005E18, NULL, this->jointTable, this->morphTable, 16);
+    Animation_Change(&this->skelAnime, sAnimations[2].animation, 1.0f, 0.0f,
+                     Animation_GetLastFrame(sAnimations[2].animation), sAnimations[2].mode, sAnimations[2].morphFrames);
+    this->previousAnimation = 2;
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    this->actor.colChkInfo.mass = 0xFF;
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Mm2/z_en_mm2.c:131]: we have defines for mass 0xFE and 0xFF now, this is `MASS_IMMOVABLE`

File: src/overlays/actors/ovl_En_Mm2/z_en_mm2.c [hunk 6]
```diff
@@ -43,33 +48,292 @@ static ColliderCylinderInit D_80AAFAC0 = {
...
+    } else {
+        func_80AAEF70(this, globalCtx);
+        this->actionFunc = func_80AAF57C;
+    }
+    if (LINK_IS_CHILD) {
+        Actor_Kill(&this->actor);
+    }
+    if (this->actor.params == 1) {
+        if (((gSaveContext.infTable[23] & 0x8000) == 0) || ((gSaveContext.eventInf[1] & 1) == 0)) {
+            osSyncPrintf(VT_FGCOL(CYAN) " マラソン 開始されていない \n" VT_RST "\n");
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Mm2/z_en_mm2.c:147]: ! on both of these flag checks

**Review by fig02 — APPROVED** [review]

File: src/overlays/actors/ovl_En_Mm2/z_en_mm2.h [hunk 7]
```diff
@@ -6,9 +6,24 @@
 
 struct EnMm2;
 
+#define SAVE_TIMER_2 ((void)0, gSaveContext.timer2Value)
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Mm2/z_en_mm2.h:9]: I personally don't think we should have a macro for this. All the `(void)0` accesses will be cleaned up eventually so I wouldn't worry about having a few instances of it in the code.

File: src/overlays/actors/ovl_En_Mm2/z_en_mm2.c [hunk 8]
```diff
@@ -43,33 +48,285 @@ static ColliderCylinderInit D_80AAFAC0 = {
...
+    }
+    return 0;
+}
+
+void func_80AAF2BC(EnMm2* this, GlobalContext* globalCtx) {
+    if (this->unk_1F6 >= 61) {
+        Actor_Kill(&this->actor);
+    }
+    SkelAnime_Update(&this->skelAnime);
+    this->unk_1F6 += 1;
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Mm2/z_en_mm2.c:171]:
```suggestion
    this->unk_1F6++;
```

File: src/overlays/actors/ovl_En_Mm2/z_en_mm2.c [hunk 9]
```diff
@@ -43,33 +48,285 @@ static ColliderCylinderInit D_80AAFAC0 = {
...
+        gSaveContext.eventInf[1] &= ~1;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mm2/EnMm2_Destroy.s")
+void func_80AAF3C0(EnMm2* this, GlobalContext* globalCtx) {
+    SkelAnime_Update(&this->skelAnime);
+    switch (this->actor.textId) {
+        case 0x607D:
+        case 0x607E: {
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Mm2/z_en_mm2.c:192]: We generally don't use braces in switch cases, so ideally they should be omitted

File: src/overlays/actors/ovl_En_Mm2/z_en_mm2.c [hunk 10]
```diff
@@ -43,33 +48,285 @@ static ColliderCylinderInit D_80AAFAC0 = {
...
+        this->actor.textId = 0x6084;
+    }
+    if (func_80AAF224(this, globalCtx, func_80AAF5EC)) {
+        this->unk_1F6 = 0;
+        if (SAVE_TIMER_2 < HIGH_SCORE(HS_MARATHON)) {
+            HIGH_SCORE(HS_MARATHON) = SAVE_TIMER_2;
+        }
+    } else {
+        LogUtils_LogThreadId("../z_en_mm2.c", 0x21D);
+        osSyncPrintf("((z_common_data.event_inf[1]) & (0x0001)) = %x\n", gSaveContext.eventInf[1] & 1);
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Mm2/z_en_mm2.c:268]:
```suggestion
        LOG_HEX("((z_common_data.event_inf[1]) & (0x0001))", gSaveContext.eventInf[1] & 1, "../z_en_mm2.c", 541);
```

File: src/overlays/actors/ovl_En_Mm2/z_en_mm2.c [hunk 11]
```diff
@@ -43,33 +48,285 @@ static ColliderCylinderInit D_80AAFAC0 = {
...
+        if ((gSaveContext.eventInf[1] & 1) == 0) {
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Mm2/z_en_mm2.c:269]:
```suggestion
        if (!(gSaveContext.eventInf[1] & 1)) {
```

File: src/overlays/actors/ovl_En_Mm2/z_en_mm2.c [hunk 12]
```diff
@@ -43,33 +48,285 @@ static ColliderCylinderInit D_80AAFAC0 = {
...
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mm2/EnMm2_Update.s")
+s32 EnMm2_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* pos, Vec3s* rot, void* thisx) {
+    EnMm2* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mm2/EnMm2_Draw.s")
+    switch (limbIndex) {
+        case 8:
+            rot->x = rot->x + this->unk_1EE.y;
+            rot->y = rot->y - this->unk_1EE.x;
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Mm2/z_en_mm2.c:315]:
```suggestion
            rot->x += this->unk_1EE.y;
            rot->y -= this->unk_1EE.x;
```

File: src/overlays/actors/ovl_En_Mm2/z_en_mm2.c [hunk 13]
```diff
@@ -43,33 +48,285 @@ static ColliderCylinderInit D_80AAFAC0 = {
...
+            break;
+        case 15:
+            rot->x = rot->x + this->unk_1E8.y;
+            rot->z = rot->z + this->unk_1E8.x + 0xFA0;
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Mm2/z_en_mm2.c:319]:
```suggestion
            rot->x += this->unk_1E8.y;
            rot->z += this->unk_1E8.x + 0xFA0;
```

File: src/overlays/actors/ovl_En_Mm2/z_en_mm2.c [hunk 14]
```diff
@@ -43,33 +48,285 @@ static ColliderCylinderInit D_80AAFAC0 = {
...
+    this->unk_1F6 += 1;
+    Math_SmoothStepToF(&this->actor.speedXZ, 10.0f, 0.6f, 2.0f, 0.0f);
+}
+
+void func_80AAF330(EnMm2* this, GlobalContext* globalCtx) {
+    if (SkelAnime_Update(&this->skelAnime)) {
+        this->actionFunc = func_80AAF2BC;
+        EnMm2_ChangeAnimation(this, 0, &this->previousAnimation);
+        this->unk_1E0 = 1;
+        if ((this->unk_1F4 & 2) == 0) {
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Mm2/z_en_mm2.c:180]:
```suggestion
        if (!(this->unk_1F4 & 2)) {
```

**Review by Roman971 — APPROVED** [review]

**Pull request merged**